### ISSUE TITLE
Adding an empty tfoot element is an a11y issue and should be avoided

### ISSUE THREAD

**Issue by Trott** [DataTables/DataTablesSrc#308]
In TablePress 3.0.1, jquery.datatables.min.js adds a `tfoot` element to a table even if the footer content is empty.

An empty `<tfoot>` is an accessibility issue (specifically, WCAG A 1.3.1: "Info and Relationships", in particular the link under "Test Rules" to [ARIA required owned elements](https://www.w3.org/WAI/standards-guidelines/act/rules/bc4a75/proposed/)).

It seems to me that https://github.com/DataTables/DataTablesSrc/blob/49e7219c8a32ba04d7954494b392276ef18115f3/js/core/core.constructor.js#L338-L340 should only run if there will be a caption. But maybe it's not possible for DataTables to know if there will be a caption or not?

Relevant table option settings screenshot:

![image](https://github.com/user-attachments/assets/03e61654-f9e7-4ded-803f-14f73fd8cd4c)

Screenshot of DOM inspector for the table after the JS appends the `<tfoot>`:

![image](https://github.com/user-attachments/assets/5dfaaebc-52cd-4df2-b9d8-878911b59430)

I'm happy to try to create a minimal DataTables-only reproduction of the issue if that's necessary, but I'm going to guess that the maintainer(s) of DataTables will have no problem doing this whereas I will have a learning curve. But if I'm wrong and I should totally create a minimal reproduction, let me know and I'll see about doing that.

Ref: https://github.com/TablePress/TablePress/issues/211

**Comment by AllanJard** [comment]
Here is the reproduction - it just needs a DataTable with no `tfoot`: https://live.datatables.net/digawozu/1/edit .

Would:

```css
tfoot:empty {
  display: none;
}
```

be an acceptable option for you? https://live.datatables.net/digawozu/2/edit .

Alternatively, I can look into removing the `tfoot` whenever the table is draws, but I'm feeling a little nervous that I'm going to miss something that would require it to be reinserted (hurrah for edge cases!).

**Comment by Trott** [comment]
> Would:
> 
> ```css
> tfoot:empty {
>   display: none;
> }
> ```
> 
> be an acceptable option for you? https://live.datatables.net/digawozu/2/edit .

My understanding is that `display: none` hides the element from the accessibility tree, so this would indeed be an acceptable solution. I'll go patch my stuff now and run a commercial accessibility assessment tool on it to confirm, because sometimes there are weird edge cases with accessibility stuff. But I think, yes, that will work.

**Comment by Trott** [comment]
I've confirmed that the Siteimprove tool agrees that the CSS you suggest resolves the issue. Thanks! Since this is CSS I can add myself, this is now resolved for me, and I'm happy to close this if you don't plan on adding the CSS to DataTablesSrc. (But it sounds to me like you plan on adding it.)

**Comment by AllanJard** [comment]
Thanks for confirming that. Agreed - I would expect `display: none` to hide the element from the accessibility tree as well, so it is great to confirm that.

I've [committed the change](https://github.com/DataTables/DataTablesSrc/commit/2f8c2558a6a68f40e2c553a63764a9a98cc3d5c8) and it will be in the next release of DataTables.
